lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1074,14 +1101,14 @@ public function resetQueryPart(string $queryPartName) : self
      */
     private function getSQLForSelect() : string
     {
-        $query = 'SELECT ' . ($this->sqlParts['distinct'] ? 'DISTINCT ' : '') .
-                  implode(', ', $this->sqlParts['select']);
+        $query = 'SELECT ' . ($this->distinct ? 'DISTINCT ' : '') .
+                  implode(', ', $this->select);
 
-        $query .= ($this->sqlParts['from'] ? ' FROM ' . implode(', ', $this->getFromClauses()) : '')
-            . ($this->sqlParts['where'] !== null ? ' WHERE ' . ((string) $this->sqlParts['where']) : '')
-            . ($this->sqlParts['groupBy'] ? ' GROUP BY ' . implode(', ', $this->sqlParts['groupBy']) : '')
-            . ($this->sqlParts['having'] !== null ? ' HAVING ' . ((string) $this->sqlParts['having']) : '')
-            . ($this->sqlParts['orderBy'] ? ' ORDER BY ' . implode(', ', $this->sqlParts['orderBy']) : '');
+        $query .= ($this->from ? ' FROM ' . implode(', ', $this->getFromClauses()) : '')
```

Comment:
I'd replace the ternary operators with a series of `if` statements for more precise code coverage.

Reply:
Done in 7a43c6a6f5c9bd3a065ea32782eba82fb42b750f.